openssl/src/x509/mod.rs

```diff
@@ -1341,6 +1595,38 @@ impl X509ReqBuilder {
         }
     }
 
+    /// Permits an attribute to be added to the certificate or certificate request.
+    pub fn add_attribute(&mut self, name: &str, value: &str) -> Result<(), ErrorStack> {
+        let name = CString::new(name).unwrap();
+        let len = value.len();
+        let value = CString::new(value).unwrap();
+        unsafe {
+            cvt(ffi::X509_REQ_add1_attr_by_txt(
+                self.0.as_ptr(),
+                name.as_ptr() as *const c_char,
+                ffi::MBSTRING_UTF8,
+                value.as_ptr() as *const c_uchar,
+                len as c_int,
+            ))
+            .map(|_| ())
+        }
+    }
+
+    pub fn add_attribute_by_nid(&mut self, nid: Nid, value: &str) -> Result<(), ErrorStack> {
```

Comment:
This pair of functions worries me, are we going to need wrappers for each type of value? Should they actually be taking an Asn1Type or Asn1String (I'm not familiar with attributes).